applications/RANSApplication/custom_elements/convection_diffusion_reaction_cross_wind_stabilized_element.h

```diff
@@ -0,0 +1,989 @@
+     * CalculateSecondDerivativesContributions,
+     * CalculateSecondDerivativesLHS, CalculateSecondDerivativesRHS methods are : OPTIONAL
+     */
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate the elemental mass matrix
+     * @param rMassMatrix: the elemental mass matrix
+     * @param rCurrentProcessInfo: the current process info instance
+     */
+    void CalculateMassMatrix(
+        MatrixType& rMassMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override
+    {
+        BoundedMatrix<double, TNumNodes, TNumNodes> local_matrix;
+        this->CalculatePrimalMassMatrix(local_matrix, rCurrentProcessInfo);
+
+        if (rMassMatrix.size1() != TNumNodes || rMassMatrix.size2() != TNumNodes) {
+            rMassMatrix.resize(TNumNodes, TNumNodes, false);
+        }
+
+        noalias(rMassMatrix) = local_matrix;
+    }
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate the elemental damping matrix
+     * @param rDampingMatrix: the elemental damping matrix
+     * @param rCurrentProcessInfo: the current process info instance
+     */
+    void CalculateDampingMatrix(
+        MatrixType& rDampingMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override
+    {
+        BoundedMatrix<double, TNumNodes, TNumNodes> local_matrix;
+        this->CalculatePrimalDampingMatrix(local_matrix, rCurrentProcessInfo);
+
+        if (rDampingMatrix.size1() != TNumNodes || rDampingMatrix.size2() != TNumNodes) {
+            rDampingMatrix.resize(TNumNodes, TNumNodes, false);
+        }
+
+        noalias(rDampingMatrix) = local_matrix;
+    }
+
+    /**
+     * This method provides the place to perform checks on the completeness of the input
+     * and the compatibility with the problem options as well as the contitutive laws selected
+     * It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input
+     * or that no common error is found.
+     * @param rCurrentProcessInfo
+     * this method is: MANDATORY
+     */
+    int Check(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        int check = BaseType::Check(rCurrentProcessInfo);
+        TConvectionDiffusionReactionData::Check(this->GetGeometry(), rCurrentProcessInfo);
+        return check;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Calculates scalar value for given gauss point
+     *
+     * @param rVariable      Scalar variable
+     * @param rShapeFunction Gauss point shape functions
+     * @param Step           Step
+     * @return double        Gauss point scalar value
+     */
+    double EvaluateInPoint(
+        const Variable<double>& rVariable,
+        const Vector& rShapeFunction,
+        const int Step = 0) const
+    {
+        return RansCalculationUtilities::EvaluateInPoint(
```

Comment:
I think this should go to the `FluidElementUtils`. This would ease the future migration.

Reply:
For sure, I have seen this reimplemented several times in different elements. :) I have asked around to have it in the core since everyone one of us needs them. But the concern they raise it having additional function calls to such a method will increase computational cost since it is being called for inner most loop at Gauss point evaluations. What do you think?

Reply:
As this won't be a `virtual` method I won't expect a prohibitive overhead. In any case we can unify the implementation in the `FluidElementUtils` for those elements that already use it (let's do it in a single PR once we merge these).

Reply:
:+1: